rdm/0004-installation-and-upgrade-workflow.md

```diff
@@ -0,0 +1,100 @@
+- Start Date: 2019-10-01
+- RFC PR: [#4](https://github.com/inveniosoftware/rfcs/pull/4)
+- Authors: Pablo Panero
+
+# Installation and upgrade workflow
+
+## Summary
+
+Overall design of the installation and upgrade workflows of Invenio RDM.
+
+## Motivation
+
+- As a developer, I want to be able to initialize the RDM file/folder structure, so that I can have a working instance.
+- As a developer, I want to lock the dependencies of my application, so that I can control them through a VCS (Version Control System) system.
+- As a developer, I want to be able to build a container image with the dependencies of my application, so I do not have to install them everytime I build the application image.
+- As a developer, I want to be able to build a container image of my application.
+- As a developer, I want to be able to run the application and the services it depends on as containers.
+- As a developer, I want to be able to run the application in a development server, so that I can easily debug and profile my application.
+- As a developer, I want to be able to stop a running instance and its corresponding services without loosing the data that is alocated in them.
+- As a developer, I want to be able to clean up the environment and all the services, so that I can start from scratch.
+- As a developer, I want to be able to update my running instance, so that I can apply changes such as customizations.
+- As a developer, I want to be able to upgrade my running instance to a newer versions of Invenio RDM.
+
+## Detailed design
+
```

Comment:
I would add a section about how it would look like for an end-user

```
$ invenio init
... <how should the output/questions look like> ...
$ invenio build
...
$ invenio run
```

Reply:
Also, how do they get the tool? (``pip install ...`` ?) And what are the requirements (Docker, Python 3, COmpose?)